hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/RpcControllerFactory.java

```diff
@@ -91,4 +107,9 @@ public static RpcControllerFactory instantiate(Configuration configuration) {
       return new RpcControllerFactory(configuration);
     }
   }
+
+  public RpcControllerFactory setRequestAttributes(Map<String, byte[]> requestAttributes) {
```

Comment:
this seems like quite the departure from master branch impl

Reply:
yeah, I'm not thrilled about it and it feels like an afterthought, but it's the least bad way that I've thought of so far. The master impl leans way more heavily into reusing controllers so just specifying on the controller is feasible. Meanwhile HTable calls `rpcControllerFactory#newController` 9 times, and that doesn't even account for multi requests. By specifying the attributes on the factory and leaning into AsyncProcess we achieve a way smaller diff than would otherwise be necessary

Reply:
I would recommend tracing how `priority` is set. 

For example, search for `newController()`, one usage is in `AsyncRequestFutureImpl`, getting passed into a MultiServerCallable (along with getPriority() val). Trace that constructor, both get passed through 3 `super()` calls, down to `RegionServerCallable`. Then in `RegionServerCallable.call(int timeout)` method, a similar pattern to what happens in AsyncTable is seen -- the controller is reset and then setPriority is called.

Most of the other non-Admin usages of `newController()` are in HTable and all are getting passed into various other subclasses of RegionServerCallable (NoncedRegionServerCallable, and ClientServiceCallable, etc). Those follow the similar pattern -- the controller and priority are passed into super calls down to RegionServerCallable.

So basically what you need to do is add `Map<String, byte[]>` as an argument to RegionServerCallable constructor, and then update all of the subclasses to pass down from their various constructors. Yea, it's a bit more files changed, but mostly 1-2 line changes in each (except for HTable which is a 1-line change per newController() call).

This might distribute the LOC to more files than your current approach, but I think it's preferable because it keeps RpcControllerFactory clean between the 2 branches. I typically try to keep classes that exist in both branches as similar as possible, and then that sometimes leads to more duplication or re-implementation in the branch-2-only classes which is not great but more maintainable over time when cherry-picking changes between branches.

Reply:
Thanks for the thoughtful reply here, that makes sense and I like the idea of preferring diffs in branch-specific classes. Ran out of time today, but will revisit this approach tomorrow

Reply:
Still need to self review, but I totally revisited the approach with this feedback in mind. thanks again!